src/NotificationPlaceholder/hooks/index.ts

```diff
@@ -0,0 +1,90 @@
+import {useEffect, useState} from 'react';
+
+declare const window: {
+    GiveNotifications: {
+        apiNonce: string;
+    };
+} & Window;
+
+interface NotificationState {
+    isLoading: boolean;
+    notifications: string[];
+}
+
+/**
+ * Fetch notifications from session storage or server
+ */
+const fetchNotifications = async () => {
```

Comment:
I feel like we should have some options for choosing retrieval from session storage, server or both - what do you think?

Reply:
I'm thinking of changing this to use `localStorage` instead of `sessionStorage`. The whole purpose of caching the notification data in `sessionStorage` is to avoid making requests to the server each time the component is loaded. 
What we might consider is having an option to select where we want to store the notification data. Currently, everything is stored in user meta, which is great for 99% of cases, but maybe we will want to have a dismissable notification that can be dismissed for all users, not just for the current one. For example, if the user did some upgrade action that can be done only once, then it doesn't make sense to show that notification to all users if action is already taken and notification is dismissed.

Reply:
I removed the fetch entirely and I'm loading data from the window object. I don't have to cache data anymore, or fetch notifications from API. This is a much cleaner and simpler solution.